### ISSUE TITLE
Timeout

### ISSUE THREAD

**Issue by igorw** [jgallen23/resistance#3]
When running in parallel, all functions are required to call the callback they receive. If they do not call it, the final callback is never reached. This is really hard to debug. It would be great if it were possible to have some timeout. When that timeout is reached, some errback is called.

I'm unsure what's the best way to design an interface for this. Adding more arguments to runParallel is most certainly a bad idea. Maybe some kind of fluent or object interface can make this more usable.

IMO it's quite an important feature. I'd be glad to help with an implementation, once a good interface is found.

**Comment by jgallen23** [comment]
I'm not sure that resistance should have that kind of control.  Resistance would have no way to "cancel" the operation you are making, so, for example, if you were saving to a DB, the action would still occur even though the timeout may have passed and resistance returned an error callback.  Let me give it a little more thought, but my initial thought is that this should be handled in your code.

**Comment by igorw** [comment]
Yeah, honestly I'm not so sure myself. It's just an issue I ran into, but perhaps it is indeed out of scope.

**Comment by igorw** [comment]
I guess I'll close this, I can't think of a generic way to handle it in resistance.
